Jellyfin.Server/Migrations/MigrationRunner.cs

```diff
@@ -111,38 +119,60 @@ private static void HandleStartupWizardCondition(IEnumerable<IMigrationRoutine>
             migrationOptions.Applied.AddRange(onlyOldInstalls.Select(m => (m.Id, m.Name)));
         }
 
-        private static void PerformMigrations(IMigrationRoutine[] migrations, MigrationOptions migrationOptions, Action<MigrationOptions> saveConfiguration, ILogger logger)
+        private static async Task PerformMigrations(
+            IMigrationRoutine[] migrations,
+            MigrationOptions migrationOptions,
+            Action<MigrationOptions> saveConfiguration,
+            ILogger logger,
+            IJellyfinDatabaseProvider? jellyfinDatabaseProvider)
         {
             // save already applied migrations, and skip them thereafter
             saveConfiguration(migrationOptions);
             var appliedMigrationIds = migrationOptions.Applied.Select(m => m.Id).ToHashSet();
+            var migrationsToBeApplied = migrations.Where(e => !appliedMigrationIds.Contains(e.Id)).ToArray();
 
-            for (var i = 0; i < migrations.Length; i++)
+            string? migrationKey = null;
+            if (migrationsToBeApplied.Any(f => f is IDatabaseMigrationRoutine) && jellyfinDatabaseProvider is not null)
             {
-                var migrationRoutine = migrations[i];
-                if (appliedMigrationIds.Contains(migrationRoutine.Id))
-                {
-                    logger.LogDebug("Skipping migration '{Name}' since it is already applied", migrationRoutine.Name);
-                    continue;
-                }
-
-                logger.LogInformation("Applying migration '{Name}'", migrationRoutine.Name);
-
+                logger.LogInformation("Database backup is performed");
```

Comment:
Make it clearer that it hasn't been done yet
```suggestion
                logger.LogInformation("Performing database backup");
```